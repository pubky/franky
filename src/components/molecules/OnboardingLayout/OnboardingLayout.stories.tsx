import type { Meta, StoryObj } from '@storybook/nextjs-vite';

import { OnboardingLayout } from './OnboardingLayout';

const meta: Meta<typeof OnboardingLayout> = {
  title: 'Molecules/Layout/OnboardingLayout',
  component: OnboardingLayout,
  parameters: {
    layout: 'fullscreen',
    docs: {
      description: {
        component:
          'A reusable layout component for onboarding pages that provides consistent structure with optional navigation.',
      },
    },
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Playground: Story = {
  args: {
    testId: 'playground-content',
    children: (
      <>
        <div className="rounded-lg bg-gray-800 p-6">
          <h1 className="mb-4 text-2xl font-bold">Page Header</h1>
          <p className="text-gray-400">This is the main content area of the template.</p>
        </div>
        <div className="rounded-lg bg-gray-800 p-6">
          <h2 className="mb-2 text-xl font-semibold">Content Section</h2>
          <p className="text-gray-400">Additional content can be added here.</p>
        </div>
      </>
    ),
  },
};

export const WithNavigation: Story = {
  args: {
    testId: 'with-nav-content',
    children: (
      <div className="rounded-lg bg-gray-800 p-6">
        <h1 className="mb-4 text-2xl font-bold">Page with Navigation</h1>
        <p className="text-gray-400">Content with navigation buttons at the bottom.</p>
      </div>
    ),
    navigation: (
      <div className="flex justify-between gap-4">
        <button className="rounded-lg bg-gray-700 px-6 py-3 hover:bg-gray-600">Back</button>
        <button className="rounded-lg bg-blue-600 px-6 py-3 hover:bg-blue-700">Next</button>
      </div>
    ),
  },
  parameters: {
    docs: {
      description: {
        story: 'Template with navigation buttons displayed at the bottom.',
      },
    },
  },
};

export const WithoutNavigation: Story = {
  args: {
    testId: 'no-nav-content',
    children: (
      <>
        <div className="rounded-lg bg-gray-800 p-6">
          <h1 className="mb-4 text-2xl font-bold">Simple Template</h1>
          <p className="text-gray-400">Template without navigation section.</p>
        </div>
        <div className="rounded-lg bg-gray-800 p-6">
          <p className="text-gray-400">Content flows naturally without bottom navigation.</p>
        </div>
      </>
    ),
  },
  parameters: {
    docs: {
      description: {
        story: 'Template without navigation component.',
      },
    },
  },
};

export const MultipleContentSections: Story = {
  args: {
    testId: 'multiple-sections',
    children: (
      <>
        <div className="rounded-lg bg-gray-800 p-6">
          <h1 className="text-2xl font-bold">Section 1</h1>
        </div>
        <div className="rounded-lg bg-gray-800 p-6">
          <h2 className="text-xl font-semibold">Section 2</h2>
        </div>
        <div className="rounded-lg bg-gray-800 p-6">
          <h3 className="text-lg">Section 3</h3>
        </div>
        <div className="rounded-lg bg-gray-800 p-6">
          <p className="text-gray-400">Section 4</p>
        </div>
      </>
    ),
    navigation: <button className="w-full rounded-lg bg-blue-600 px-6 py-3 hover:bg-blue-700">Continue</button>,
  },
  parameters: {
    docs: {
      description: {
        story: 'Template with multiple content sections and single navigation button.',
      },
    },
  },
};

export const BackupPageExample: Story = {
  args: {
    testId: 'backup-content',
    children: (
      <>
        <div className="rounded-lg bg-gray-800 p-6">
          <h1 className="mb-4 text-2xl font-bold">Backup Your Account</h1>
          <p className="text-gray-400">Choose your preferred backup method.</p>
        </div>
        <div className="rounded-lg bg-gray-800 p-6">
          <div className="space-y-4">
            <button className="w-full rounded-lg border border-gray-700 p-4 text-left hover:bg-gray-700">
              Backup with Recovery Phrase
            </button>
            <button className="w-full rounded-lg border border-gray-700 p-4 text-left hover:bg-gray-700">
              Backup with File
            </button>
          </div>
        </div>
      </>
    ),
    navigation: (
      <div className="flex justify-between gap-4">
        <button className="rounded-lg bg-gray-700 px-6 py-3 hover:bg-gray-600">Back</button>
        <button className="rounded-lg bg-blue-600 px-6 py-3 hover:bg-blue-700">Continue</button>
      </div>
    ),
  },
  parameters: {
    docs: {
      description: {
        story: 'Example showing how the Backup template uses this layout component.',
      },
    },
  },
};
