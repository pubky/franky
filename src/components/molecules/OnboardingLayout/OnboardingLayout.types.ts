export interface OnboardingLayoutProps {
  /**
   * Unique identifier for the content container
   */
  testId: string;
  /**
   * Main content to be displayed
   */
  children: React.ReactNode;
  /**
   * Optional navigation component to be displayed at the bottom
   */
  navigation?: React.ReactNode;
}
