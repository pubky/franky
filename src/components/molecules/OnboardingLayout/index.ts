export { OnboardingLayout } from './OnboardingLayout';
export type { OnboardingLayoutProps } from './OnboardingLayout.types';
