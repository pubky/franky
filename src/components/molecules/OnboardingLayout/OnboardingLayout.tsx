import * as Atoms from '@/atoms';

import { OnboardingLayoutProps } from './OnboardingLayout.types';

export function OnboardingLayout({ testId, children, navigation }: OnboardingLayoutProps) {
  return (
    <Atoms.Container
      size="container"
      className="min-h-dvh items-stretch gap-0 px-6 pb-0 pt-4 lg:min-h-0 lg:items-start lg:pb-6"
    >
      <div data-testid={testId} className="flex w-full flex-1 flex-col gap-0 lg:flex-none">
        {children}
      </div>
      {navigation && <div className="onboarding-nav mt-auto w-full lg:mt-0">{navigation}</div>}
    </Atoms.Container>
  );
}
