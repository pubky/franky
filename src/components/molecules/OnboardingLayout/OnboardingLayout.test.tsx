import React from 'react';
import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { OnboardingLayout } from './OnboardingLayout';

describe('OnboardingLayout', () => {
  it('renders children correctly', () => {
    render(
      <OnboardingLayout testId="test-content">
        <div>Test Content</div>
      </OnboardingLayout>,
    );

    expect(screen.getByTestId('test-content')).toBeInTheDocument();
    expect(screen.getByText('Test Content')).toBeInTheDocument();
  });

  it('renders with navigation when provided', () => {
    render(
      <OnboardingLayout testId="with-nav" navigation={<button>Next</button>}>
        <div>Content</div>
      </OnboardingLayout>,
    );

    expect(screen.getByRole('button', { name: 'Next' })).toBeInTheDocument();
  });

  it('renders without navigation when not provided', () => {
    const { container } = render(
      <OnboardingLayout testId="no-nav">
        <div>Content</div>
      </OnboardingLayout>,
    );

    const navContainer = container.querySelector('.onboarding-nav');
    expect(navContainer).not.toBeInTheDocument();
  });
});

describe('OnboardingLayout - Snapshots', () => {
  it('matches snapshot with only content', () => {
    const { container } = render(
      <OnboardingLayout testId="content-only">
        <div>Simple Content</div>
      </OnboardingLayout>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with content and navigation', () => {
    const { container } = render(
      <OnboardingLayout testId="with-navigation" navigation={<button>Next Step</button>}>
        <div>Content with Navigation</div>
      </OnboardingLayout>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with multiple children', () => {
    const { container } = render(
      <OnboardingLayout testId="multiple-children">
        <header>Header</header>
        <main>Main Content</main>
        <footer>Footer</footer>
      </OnboardingLayout>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with complex navigation', () => {
    const { container } = render(
      <OnboardingLayout
        testId="complex-nav"
        navigation={
          <div className="flex gap-4">
            <button>Back</button>
            <button>Next</button>
          </div>
        }
      >
        <div>Content</div>
      </OnboardingLayout>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
