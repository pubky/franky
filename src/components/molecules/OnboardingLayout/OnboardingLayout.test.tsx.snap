// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OnboardingLayout - Snapshots > matches snapshot with complex navigation 1`] = `
<div
  class="mx-auto w-full flex-col flex container min-h-dvh items-stretch gap-0 px-6 pb-0 pt-4 lg:min-h-0 lg:items-start lg:pb-6"
  data-testid="container"
>
  <div
    class="flex w-full flex-1 flex-col gap-0 lg:flex-none"
    data-testid="complex-nav"
  >
    <div>
      Content
    </div>
  </div>
  <div
    class="onboarding-nav mt-auto w-full lg:mt-0"
  >
    <div
      class="flex gap-4"
    >
      <button>
        Back
      </button>
      <button>
        Next
      </button>
    </div>
  </div>
</div>
`;

exports[`OnboardingLayout - Snapshots > matches snapshot with content and navigation 1`] = `
<div
  class="mx-auto w-full flex-col flex container min-h-dvh items-stretch gap-0 px-6 pb-0 pt-4 lg:min-h-0 lg:items-start lg:pb-6"
  data-testid="container"
>
  <div
    class="flex w-full flex-1 flex-col gap-0 lg:flex-none"
    data-testid="with-navigation"
  >
    <div>
      Content with Navigation
    </div>
  </div>
  <div
    class="onboarding-nav mt-auto w-full lg:mt-0"
  >
    <button>
      Next Step
    </button>
  </div>
</div>
`;

exports[`OnboardingLayout - Snapshots > matches snapshot with multiple children 1`] = `
<div
  class="mx-auto w-full flex-col flex container min-h-dvh items-stretch gap-0 px-6 pb-0 pt-4 lg:min-h-0 lg:items-start lg:pb-6"
  data-testid="container"
>
  <div
    class="flex w-full flex-1 flex-col gap-0 lg:flex-none"
    data-testid="multiple-children"
  >
    <header>
      Header
    </header>
    <main>
      Main Content
    </main>
    <footer>
      Footer
    </footer>
  </div>
</div>
`;

exports[`OnboardingLayout - Snapshots > matches snapshot with only content 1`] = `
<div
  class="mx-auto w-full flex-col flex container min-h-dvh items-stretch gap-0 px-6 pb-0 pt-4 lg:min-h-0 lg:items-start lg:pb-6"
  data-testid="container"
>
  <div
    class="flex w-full flex-1 flex-col gap-0 lg:flex-none"
    data-testid="content-only"
  >
    <div>
      Simple Content
    </div>
  </div>
</div>
`;
